### DISCUSSION TITLE
Is it possible to make it multiple date picker instead of range picker?

### DISCUSSION THREAD

**Discussion by amarg26** [dmuy/duDatepicker#18] [Q&A] [answered]
Hi Author,

Your plugin is super simple and amazing. I want to change this selecting range dates to single selecting multiple dates. Is this feature available or it will be too big thing to develop??

Also can we add customization at bottom like custom Html buttons and bind the event functions like click etc. 

Thanks, Take Care.

**Comment by dmuy** [comment, marked as answer]
Hi @amarg26,

Selecting multiple dates is a good feature, maybe I'll add this in the future.

For the customized button, which button is this? Can you elaborate on this?

Thanks

**Reply by amarg26** [reply]
Hi @dmuy ,

Customized button means additional buttons we want to add such as clear, cancel reset, etc. and firing click event. 

One more thing.

After using it for mobile devices and testing it on actual android and ios.
The date picker position is not the center of the screen on android and ios.
It is too big in size. It cuts the mobile screen at the bottom and ok, clear buttons are inaccessible.

Actual tested on iPhone 7, Honor 7x, OnePlus 8, Redmi 9. It might be the same behavior for other devices also. 
If you want screenshots, video shorts. I am ready to provide you.

I request you to kindly reproduce it on actual devices.

Thanks, Stay Safe.

**Reply by dmuy** [reply]
Hi @amarg26,

The datepicker already has OK, Cancel, and Clear buttons:
```javascript
// Determines if clicking the date will automatically select it; OK button will not be displayed if true
auto: false,  
// Determines if Clear button is displayed
clearBtn: false,      
// Determines if Cancel button is displayed
cancelBtn: false,     
```
I don't see the point of allowing to add more buttons to it, maybe an option to allow custom content of the button (i.e. adding html to add icons, etc), because it will somewhat compromise the overall design of it. Also, OK, Cancel and Clear click handler callback can be implemented in the future.

Lastly, for the mobile sizing issue, can you provide a screenshot (of GIF) or a video of how it behaves? I only tested this using the device toolbar on chrome. Sadly I cannot test this on actual phones because I don't have any of those you mentioned.

Thanks

**Reply by amarg26** [reply]
Mobile screenshot - iPhone 7 portrait & Landscape.

![Screenshot_20210420_095820_com huawei himovie overseas](https://user-images.githubusercontent.com/22127481/115338087-ca500780-a1bf-11eb-883d-7000aae8e9f2.jpg)
![Screenshot_20210420_095732_com huawei himovie overseas](https://user-images.githubusercontent.com/22127481/115338101-ce7c2500-a1bf-11eb-948f-524ccd68a282.jpg)

**Reply by dmuy** [reply]
Hi @amarg26,

All I can say based on these screenshots are: 

1. For the portrait, it's like there's a scrollable horizontal content to the right of the website itself (if you get what I mean).
2. For landscape, I have no idea why it's like it is zoomed in. iPhone 7 has a big screen resolution so I don't know why it behaves this way. 
 
Is it possible for me to test the website itself?

**Reply by amarg26** [reply]
Yes, I have fixed zooming problems using meta tag zoom property. It was old screenshot. But I have confirmed by mobile testers that even after that position of datepicker modal was not on center of the screen on both devices. I can only provide you screenshots only. since it is used for company project. As per policy can't add you in white list.   I would like to suggest you to deploy this datepicker to heroku app then you can test it on actual devices.
